build-advisories/src/Roave/SecurityAdvisories/VersionConstraint.php

```diff
@@ -196,4 +248,92 @@ private function containsUpperBound($otherUpperBoundIncluded, Version $otherUppe
 
         return $this->upperBound->isGreaterThan($otherUpperBound);
     }
+
+    /**
+     * @param VersionConstraint $other
+     *
+     * @return bool
+     */
+    private function overlapsWith(VersionConstraint $other)
+    {
+        if (! $this->isSimpleRangeString() && $other->isSimpleRangeString()) {
+            return false;
+        }
+
+        if ($this->contains($other) || $other->contains($this)) {
+            return false;
+        }
+
+        $containsLower = $this->strictlyContainsOtherBound($other->lowerBound);
+        $containsUpper = $this->strictlyContainsOtherBound($other->upperBound);
+
+        return $containsLower xor $containsUpper;
+
+        return $this->strictlyContainsOtherBound($other->lowerBound)
+            xor $this->strictlyContainsOtherBound($other->upperBound);
+    }
+
+    /**
+     * @param VersionConstraint $other
+     *
+     * @return bool
+     *
+     * @throws \LogicException
+     */
+    private function mergeWithOverlapping(VersionConstraint $other)
+    {
+        if (! $this->overlapsWith($other)) {
+            throw new \LogicException(sprintf(
+                '%s "%s" does not overlap with %s "%s"',
+                get_class($this),
```

Comment:
not needed here either, as the class is also known at compile time as it is final